File: app/src/main/java/com/mytlogos/enterprise/background/api/ExternalUserApi.java
package com.mytlogos.enterprise.background.api;

import com.mytlogos.enterprise.background.api.model.ClientExternalUser;

import java.util.List;
import java.util.Map;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;
import retrofit2.http.Url;

interface ExternalUserApi {

    @GET
    Call<ClientExternalUser> getExternalUser(@Url String url, @QueryMap Map<String, Object> body);

    @GET
    Call<List<ClientExternalUser>> getExternalUsers(@Url String url, @QueryMap Map<String, Object> body);

    @POST
    Call<ClientExternalUser> addExternalUser(@Url String url, @Body Map<String, Object> body);

    @DELETE
    Call<Boolean> deleteExternalUser(@Url String url, @Body Map<String, Object> body);
}

File: app/src/main/java/com/mytlogos/enterprise/background/api/ExternalUserApi.kt
package com.mytlogos.enterprise.background.api

import com.mytlogos.enterprise.background.api.model.ClientExternalUser
import retrofit2.Response
import retrofit2.http.*

internal interface ExternalUserApi {
    @GET("{start}/refresh")
    suspend fun  refreshExternalUser(@Path(value = "start", encoded = true) url: String, @QueryMap body: MutableMap<String, Any?>): Response<ClientExternalUser>

    @GET
    suspend fun  getExternalUser(@Url url: String, @QueryMap body: MutableMap<String, Any?>): Response<ClientExternalUser>

    @GET
    suspend fun  getExternalUsers(@Url url: String, @QueryMap body: MutableMap<String, Any?>): Response<List<ClientExternalUser>>

    @POST
    suspend fun  addExternalUser(@Url url: String, @Body body: MutableMap<String, Any?>): Response<ClientExternalUser>

    @DELETE
    suspend fun  deleteExternalUser(@Url url: String, @Body body: MutableMap<String, Any?>): Response<Boolean>
}